fvm/evm/impl/impl.go

```diff
@@ -648,21 +649,30 @@ func newInternalEVMTypeWithdrawFunction(
 				panic(errors.NewUnreachableError())
 			}
 
-			amount := types.NewBalance(amountValue.BigInt)
+			_, overflow := uint256.FromBig(amountValue.BigInt)
+			if overflow {
+				panic(types.ErrInvalidBalance)
+			}
+
+			if !types.AttoFlowBalanceValidForFlowVault(amountValue.BigInt) {
+				panic(types.ErrWithdrawBalanceRounding)
+			}
+
+			value := new(big.Int).Div(amountValue.BigInt, types.UFixToAttoConversionMultiplier)
+			amount := types.NewBalanceFromUFix64(cadence.UFix64(value.Uint64()))
 
 			// Withdraw
 
 			const isAuthorized = true
 			account := handler.AccountByAddress(fromAddress, isAuthorized)
 			vault := account.Withdraw(amount)
 
-			ufix, roundedOff, err := types.ConvertBalanceToUFix64(vault.Balance())
+			// We have already truncated the remainder above, so we do not
+			// care about rounding loss here.
+			ufix, _, err := types.ConvertBalanceToUFix64(vault.Balance())
 			if err != nil {
 				panic(err)
 			}
-			if roundedOff {
```

Comment:
I know this can now not happen, but I would leave the check in as a redundancy.

Reply:
Added back in https://github.com/onflow/flow-go/pull/6877/commits/aed426cf139486af5d0e28f5bc67ec73f4242287